lib/email_auth.py

```diff
@@ -0,0 +1,62 @@
+# email_auth.py
+#
+# Combine standard login with option of OAuth2.0
+#
+
+import smtplib
+import base64
+from httplib2 import Http
+from oauth2client import file, client, tools
+from googleapiclient.discovery import build
+from lib.config import open_config
+from imaplib import IMAP4_SSL
+
+config = open_config()
+email_config = config['email']
+gmail_url = "https://mail.google.com/"
+imapUrl = "imap.gmail.com"
+smtpUrl = "smtp.gmail.com"
+smtpPort = "587"
+username= email_config["username"]
+
+
+def email_authentication():
+   
+  mail = IMAP4_SSL(imapUrl)
+  if "password"  in email_config and email_config["password"]:
+    mail.login(email_config['username'], email_config['password'])
+  else:
+    creds = get_oauth_credentials(gmail_url)
+    client_credentials = creds.get_access_token().access_token
+    authstring = f"user={username}\1auth=Bearer {client_credentials}\1\1"
+    mail.authenticate('XOAUTH2', lambda x: authstring)
+  return mail
+
+
+def send_email(self, recipients, message):
```

Comment:
we should remove all the "self" references here since we're not in a class

Reply:
fixed